src/wp-includes/compat.php

```diff
@@ -374,7 +374,42 @@ function is_iterable( $var ) {
 		return ( is_array( $var ) || $var instanceof Traversable );
 	}
 }
+if ( ! function_exists( 'array_key_first' ) ) {
+	/**
+	 * Polyfill for array_key_first() function added in PHP 7.3.
+	 *
+	 * Get the first key of the given array without affecting the internal array pointer.
+	 *
+	 * @ticket 45055
+	 * @ignore
+	 * @since 5.9.0
+	 *
+	 * @param array $arr An array.
+	 * @return string|int|null The first key of array if the array is not empty. NULL otherwise.
+	 */
+	function array_key_first( array $arr ) {
+		foreach ( $arr as $key => $value ) {
+				return $key;
+		}
+	}
+}
 
+if ( ! function_exists( 'array_key_last' ) ) {
+	/**
+	 * Polyfill for array_key_last() function added in PHP 7.3.
+	 *
+	 * Get the last key of the given array without affecting the internal array pointer.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param array $arr An array.
+	 * @return string|int|null The first key of array if the array is not empty. NULL otherwise.
+	 */
+	function array_key_last( array $arr ) {
+			end( $arr );
```

Comment:
`end()` advances the array's internal point. https://www.php.net/manual/en/function.end.php

>end() advances array's internal pointer to the last element, and returns its value.

Reply:
https://3v4l.org/XjTn8#v5.6.40

Reply:
After discussing with @costdev, I was wrong. `end()` within the function does change the internal point, however, it's changing it on the `$arr` which is in function scope. As the function's parameter is not passed by reference, it does not affect the originally array's point. Doh on my part.

Reply:
Reverted in https://github.com/WordPress/wordpress-develop/pull/1809/commits/68893cf8c893f863186d17d03b014e10db92b910. But does retain the `empty()` check.